src/inmanta/server/agentmanager.py

```diff
@@ -668,12 +694,42 @@ async def list_agent_processes(self, environment: Optional[UUID], expired: bool)
         return 200, {"processes": processes}
 
     @protocol.handle(methods.list_agents, env="tid")
-    async def list_agents(self, env: Optional[data.Environment]) -> Apireturn:
+    async def list_agents(
+        self,
+        env: Optional[data.Environment],
+        start: Optional[str] = None,
+        end: Optional[str] = None,
+        limit: Optional[int] = None,
+    ) -> Apireturn:
+        """List all agents whose name is after start and before end
+
+        :param env: The environment the agents should come from
+        :param start: The name all of the selected agent should be greater than this, defaults to None
+        :param end: The name all of the selected agent should be smaller than this, defaults to None
+        :param limit: Whether to limit the number of returned entries, defaults to None
+        :raises BadRequest: Limit, start and end can not be set together
```

Comment:
This comment should be removed.

Reply:
Did it in https://github.com/inmanta/inmanta/pull/2521/commits/541aebaff4c0a488a53e0082c7783e45da04f8d6